packages/app/src/elements/audit.ts

```diff
@@ -0,0 +1,214 @@
+        return html`
+            <div class="padded layout">
+                <div class="vertical spacing">
+                    <h3>
+                        Here you can see some counts for passwords you should change, as they might be easier to be
+                        easily guessed.
+                    </h3>
+                    <p class="subtle">Click a count to view the affected items.</p>
+                    <div class="counts">
+                        <pl-audit-list-item
+                            .listItems=${_reusedPasswords}
+                            .label=${$l("Reused Passwords")}
+                            .routeParams=${{ auditReused: "true" }}
+                        ></pl-audit-list-item>
+                        <pl-audit-list-item
+                            .listItems=${_weakPasswords}
+                            .label=${$l("Weak Passwords")}
+                            .routeParams=${{ auditWeak: "true" }}
+                        ></pl-audit-list-item>
+                        <pl-audit-list-item
+                            .listItems=${_compromisedPasswords}
+                            .label=${$l("Compromised Passwords")}
+                            .routeParams=${{ auditCompromised: "true" }}
+                        ></pl-audit-list-item>
+                    </div>
+                </div>
+            </div>
+        `;
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        this.addEventListener("change", () => this.audit());
+    }
+
+    async audit() {
+        this._reusedPasswords = [];
+        this._weakPasswords = [];
+        this._compromisedPasswords = [];
+
+        const { vaults } = this.state;
+
+        // This prevents the same item from being considered more than once for different sections/audits
+        const checkedItemIds: Set<string> = new Set();
+
+        for (const vault of vaults) {
+            for (const item of vault.items) {
+                if (checkedItemIds.has(item.id)) {
+                    continue;
+                }
+
+                // TODO: Skip/ignore if the item doesn't have a password field
+
+                // TODO: Actually check if the audit is correct, instead of this random calculation
+                const randomNumber = Math.floor(Math.random() * 2);
+
+                if (randomNumber % 5 === 0) {
+                    this._reusedPasswords.push({
+                        vault,
+                        item,
+                    });
+
+                    checkedItemIds.add(item.id);
+                } else if (randomNumber % 5 === 1) {
+                    this._weakPasswords.push({
+                        vault,
+                        item,
+                    });
+
+                    checkedItemIds.add(item.id);
+                } else if (randomNumber % 5 === 2) {
+                    this._compromisedPasswords.push({
+                        vault,
+                        item,
+                    });
```

Comment:
This is where you're going wrong. Reactive properties are implemented using simple getters and setters, and don't pick up on manipulations of complex objects. I.e., in order to trigger an update, you have to update the object (reference) itself. So the pattern you should use looks something like this:

```ts
const reused = [];
const weak = [];
const compromised = [];

// Fill arrays...

this._reusedPasswords = reused;
this._weakPasswords = weak;
this._compromisedPasswords = compromised;
```

Reply:
Alternatively, you *can* trigger updates manually as well, using `requestUpdate`. E.g.:

```ts
this._reusedPasswords.push({vault, item});
this.requestUpdate();
```

See also: https://lit.dev/docs/components/lifecycle/#reactive-update-cycle-triggering